### DISCUSSION TITLE
How can I refetch a data after mutation?

### DISCUSSION THREAD

**Discussion by lgj9172** [TanStack/query#3665] [Q&A] [answered]
I want to refetch a data using mutation result.
(task: make a model in sever and get upload url using model id)

So what I done is follow:
1.  I set a enabled false at useQuery
2. Do useMutation
3. Get a data from mutation and set value
4. Do refetch in useQuery

But it is not work because set value and refetch is unsynchronous
How can I do refetch manually using mutation result?

```
  const [modelID, setModelID] = useState("");
  const GetModelUploadURL = useQuery(
    ["uploadURL", modelID],
    () => getModelUploadURL({ modelId: modelID }),
    {
      enabled: false,
      onSuccess: ({ data }) => {
        console.log(data);
      },
    }
  );
  const PostModel = useMutation((model) => postModel(model), {
    onSuccess: ({ data }) => {
      console.log(data);
      setModelID(data.model_id);
      GetModelUploadURL.refetch();
    },
    onError: ({ data }) => {
      console.log(data);
    },
  });
```

**Comment by TkDodo** [comment, marked as answer]
this is literally the same question as:

- https://github.com/tannerlinsley/react-query/discussions/3664

so again:
- set `enabled` to `!!modelID`
- don't call `refetch`, just call `setModelID`
- see the magic happen

**Reply by lgj9172** [reply]
OMG It works!
What is meaning of operator (!!) ?
I thought this `!!true == true`

**Reply by TkDodo** [reply]
it coerces a value to Boolean. If it's `undefined` or `null`, it becomes `false`. If it's the number `7`, it becomes `true`

**Reply by lgj9172** [reply]
What a beutiful skill! Thank you sir. I love you.
